### PR TITLE
Fix EZP-24581: Creating a new content type failed with error: Failed …

### PR DESCRIPTION
**Pull request by glye**
…to load '/contenttype/create/1'

Followup to platformui fix, adds trait for isNew()
https://github.com/ezsystems/PlatformUIBundle/pull/291

https://jira.ez.no/browse/EZP-24581

### CODE REVIEW

File: lib/Data/ContentTypeData.php [hunk 1]
```diff
@@ -31,6 +31,16 @@ class ContentTypeData extends ContentTypeUpdateStruct
      */
     protected $fieldDefinitionsData = [];
 
+    /**
+     * Trait which provides isNew(), and mandates getIdentifier().
+     */
+    use RepositoryFormData;
```
Comment (lolautruche) [inline lib/Data/ContentTypeData.php]: Please add traits `use` statements before properties.

File: lib/Data/RepositoryFormData.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * @version //autogentag//
+ */
+
+namespace EzSystems\RepositoryForms\Data;
+
+/**
+ * Trait for repository data objects.
+ */
+trait RepositoryFormData
```
Comment (lolautruche) [inline lib/Data/RepositoryFormData.php]: Not sure about the name here, especially about `Form` inside. This has nothing to do with a form.

File: lib/Data/RepositoryFormData.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    /**
+     * Whether the Data object is considered new, based on the identifier
+     * If it isn't new, one can e.g. use the identifier from the underlying value object.
+     */
+    public function isNew()
+    {
+        return strpos($this->getIdentifier(), '__new__') === 0;
+    }
+
+    abstract public function getIdentifier();
```
Comment (lolautruche) [inline lib/Data/RepositoryFormData.php]: Needs PHPDoc

File: lib/Data/RepositoryFormData.php (hunk 2 above)
Comment (glye) [inline lib/Data/RepositoryFormData.php]: @lolautruche Hm... `IsNewTrait`? uh... `NoveltyRecognizerTrait`? Yup, naming is hard :)

Comment (bdunogier) [inline lib/Data/RepositoryFormData.php]: `freshnessChecker` ?

Comment (glye) [inline lib/Data/RepositoryFormData.php]: Thanks. As much as I like _sleuth_, _freshness_, and the formality of `NoveltyAssessor`, I think I'll suggest the plain `NewnessChecker`.

**Comment by glye** [comment]
Moved trait use statement, changed trait name to NewnessChecker, added phpdoc

**Comment by glye** [comment]
Rebased on master to get CS fixes in.

**Comment by andrerom** [comment]
+1

**Comment by lolautruche** [comment]
+1

**Pull request merged**